src/SmartHotel.Clients/SmartHotel.Clients/Services/IoT/RoomDevicesDataService.cs

```diff
@@ -45,16 +51,36 @@ public async Task<RoomTemperature> GetRoomTemperatureAsync()
 				}
 				else
 				{
-					//TODO: Handle the case where we haven't actually retrieved data yet
-					// simulates values returned for IoT device
-					var currentTemp = 70;
-					var desiredTemp = 76;
-					return new RoomTemperature( new SensorValue( currentTemp ), new SensorValue( desiredTemp ) );
+                    var roomData = await RoomSensorData(token, _roomId);
```

Comment:
This logic also needs to be called in the **SensorDataPollingTimerTick** method. Otherwise that tick isn't putting anything in **_currentSensorDataBySensorDataType**

AND the token to pass into ALL usages of this method is the **_authenticationService.AuthenticatedUser.Token**